### ISSUE TITLE
Pre Pump Detection

### ISSUE THREAD

**Issue by ernestonee** [fj317/PumpBot#12]
Hey we all know that Big Groups who make the pump Signals are buying the coins before everyone Else, and since your bots scans the coins and detetects if they are pre pumped, maybe you could add a Feature where it tells you which coin went up in volume for example in the last 30 minutes

**Comment by fj317** [comment]
Yes this would be a nice feature but it'll take some time and be difficult to implement. I'll add it to the future feature section.

**Comment by TaylorCCMG** [comment]
I'd say the better option would be the auto-pump detection. The bot scans all coins every 1 second, looking for price changes, and automatically buys the one that's being pumped based on the price change. So the order will be placed at 21:00:02, not 21:00:05, when you enter coin manually. Seconds matter!

**Comment by OjusWiZard** [comment]
That's a great idea and I was searching about this. It has already been researched, and they created a machine learning algorithm that predicts the pump signal based on many factors.
I skimmed through it but could understand their code much because it's not in python.
[The Anatomy of a Cryptocurrency Pump-and-Dump Scheme](https://arxiv.org/abs/1811.10109)

**Comment by fj317** [comment]
> That's a great idea and I was searching about this. It has already been researched, and they created a machine learning algorithm that predicts the pump signal based on many factors.
> I skimmed through it but could understand their code much because it's not in python.
> [The Anatomy of a Cryptocurrency Pump-and-Dump Scheme](https://arxiv.org/abs/1811.10109)

This looks really interesting. I'll take a more detailed look when I have more time. Looks complicated and uses R as the language which I have minimal experience in using so may be hard to port that exact code.

**Comment by lbosk** [comment]
This machine learning is a different matter: they look to the trades in the previous day, looking for patterns that suggest that someone is accumulating coins. It looks very difficult to implement.
On the other side, scanning the market at high rate during the previous minute, looking for rapid rises of price, seems a good idea. I'll have a look to this....

**Comment by fj317** [comment]
> This machine learning is a different matter: they look to the trades in the previous day, looking for patterns that suggest that someone is accumulating coins. It looks very difficult to implement.
> On the other side, scanning the market at high rate during the previous minute, looking for rapid rises of price, seems a good idea. I'll have a look to this....

I feel like if you wanted to implement something like this you might want to start your own repo with a new bot since it would be a slightly different concept to the current pump & dump bot. Although I would happily implement your changes if you believe you could add them to the codebase in a nice way.

**Comment by brianleect** [comment]
> Yes this would be a nice feature but it'll take some time and be difficult to implement. I'll add it to the future feature section.

I happened to come across your project and I believe a recent project I was working on would solve this issue. It is able to check prices of pairs on binance for up to 1s intervals and sends a telegram message to a group if the condition for a pump is met.

https://github.com/brianleect/binance-pump-alerts/
